compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -590,6 +594,20 @@ pub fn compile_declarative_macro(
     (mk_syn_ext(expander), rule_spans)
 }
 
+fn annotate_doc_comment(
+    err: &mut DiagnosticBuilder<'_, ErrorGuaranteed>,
+    sm: &SourceMap,
+    span: Span,
+) {
+    if let Ok(src) = sm.span_to_snippet(span) {
+        if src.starts_with("///") || src.starts_with("/**") {
+            err.span_label(span, "outer doc comments expand to `#[doc = \"...\"]`, which is what this macro attempted to match");
```

Comment:
nit: you could make these `SessionSubdiagnostic`s or otherwise translatable with typed identifiers

Reply:
can do!